src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -3349,6 +3449,330 @@ public <T> T invokeAny(Collection<? extends Callable<T>> tasks,
+     * <li>An execution or the task encounters resource exhaustion.
+     * <li>An execution of the task throws an exception.  In this case
+     * calling {@link Future#get() get} on the returned future will throw
+     * {@link ExecutionException}, holding the exception as its cause.
+     * </ul>
+     * Subsequent executions are suppressed.  Subsequent calls to
+     * {@link Future#isDone isDone()} on the returned future will
+     * return {@code true}.
+     * @param command the task to execute
+     * @param initialDelay the time to delay first execution
+     * @param delay the delay between the termination of one
+     * execution and the commencement of the next
+     * @param unit the time unit of the initialDelay and delay parameters
+     * @return a ForkJoinTask implementing the ScheduledFuture
+     *         interface.  The future's {@link Future#get() get()}
+     *         method will never return normally, and will throw an
+     *         exception upon task cancellation or abnormal
+     *         termination of a task execution.
+     * @throws RejectedExecutionException if the pool is shutdown or
+     *         submission encounters resource exhaustion.
+     * @throws NullPointerException if command or unit is null
+     * @throws IllegalArgumentException if delay less than or equal to zero
+     * @since 25
+     */
+    public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command,
+                                                     long initialDelay,
+                                                     long delay, TimeUnit unit) {
+        Objects.requireNonNull(command);
+        if (delay <= 0L)
+            throw new IllegalArgumentException();
+        return scheduleDelayedTask(
+            new ScheduledForkJoinTask<Void>(
+                unit.toNanos(initialDelay),
+                -unit.toNanos(delay),  // negative for fixed delay
+                false, command, null, this));
+    }
+
+    /**
+     * Body of a task performed on timeout of another task
+     */
+    static final class TimeoutAction<V> implements Runnable {
+        // set after construction, nulled after use
+        ForkJoinTask.CallableWithTimeout<V> task;
+        Consumer<ForkJoinTask<V>> action;
+        TimeoutAction(Consumer<ForkJoinTask<V>> action) {
+            this.action = action;
+        }
+        public void run() {
+            ForkJoinTask.CallableWithTimeout<V> t = task;
+            Consumer<ForkJoinTask<V>> a = action;
+            task = null;
+            action = null;
+            if (t != null && t.status >= 0) {
+                if (a == null)
+                    t.cancel(true);
+                else {
+                    a.accept(t);
+                    t.interruptIfRunning(true);
+                }
+            }
+        }
+    }
+
+    /**
+     * Submits a task executing the given function, cancelling or
+     * performing a given timeoutAction if not completed within the
```

Comment:
"cancelling or performance a given timeoutAction" could be misread to mean that it cancels the timeoutAction. It might be clear to say "cancelling the task or ...".

Reply:
Added:
Delayed
 * actions become <em>enabled</em> and behave as ordinary submitted
 * tasks when their delays elapse.